### ISSUE TITLE
Intl Component Directory Traversal/Script Execution via $locale Value

### ISSUE THREAD

**Issue by dbohannon** [symfony/symfony#24056]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | yes
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.3

The Intl component includes various bundle readers (e.g. PhpBundleReader, JsonBundleReader, etc.) used to read resource bundles from the local filesystem. The read() APIs for these classes accept a $path argument to the directory where the bundles are located and a $locale argument specifying which language bundle to retrieve. The read() API simply appends the $locale string to the $path string, adds the appropriate suffix (e.g. .php, .json, etc.). and returns the data from the file. 

The $locale value is commonly retrieved from the untrusted query string and, if not validated/sanitized, will lead to directory traversal attacks as an attacker can navigate to arbitrary directories via the dot-dot-slash attack. This is especially problematic when using the PhpBundleReader since the PHP file is executed via an include statement, allowing an attacker to execute arbitrary PHP files on the system.

Consider the example excerpt below demonstrating a vulnerable example where the $locale retrieved from the query can be used to execute arbitrary PHP files.

```php
use Symfony\Component\Intl\Data\Bundle\Reader\PhpBundleReader;  
...  
//get locale from query string, default to 'en'  
$locale = $request->query->get('locale');  
if($locale === null){  
    $locale = 'en';  
}  
//read intl bundle from file  
$reader = new PhpBundleReader();  
$intl_data = $reader->read('./bundles', $locale);  
...
```

Optimally, the read() APIs should sanitize the $locale argument to prevent directory traversal attacks. At a minimum, a warning should be  added to the Intl documentation stating that the $locale will be appended to the $path value and should be sanitized appropriately to prevent directory traversal attacks.

**Comment by javiereguiluz** [comment]
@dbohannon thanks for your report! I don't know if this is a serious security issue or not, but next time, please consider using this channel to report security vulnerabilities: https://symfony.com/security

Thanks!

**Comment by xabbuh** [comment]
fixed in symfony/symfony#24994
